plutus-core/untyped-plutus-core/Language/UntypedPlutusCore/Core/Type.hs

```diff
@@ -1,7 +1,17 @@
-{-# LANGUAGE FlexibleInstances #-}
-{-# LANGUAGE TypeFamilies      #-}
+{-# LANGUAGE FlexibleInstances    #-}
+{-# LANGUAGE TypeFamilies         #-}
+{-# LANGUAGE TypeOperators        #-}
+{-# LANGUAGE UndecidableInstances #-}
 
-module Language.UntypedPlutusCore.Core.Type where
+module Language.UntypedPlutusCore.Core.Type
```

Comment:
I think I mentioned this before, but do we really want this to be called `Type`?  I always assumed that the corresponding module in the typed version was called `Type` because it defined the `Type` type, but that doesn't make much sense here.  Maybe it was called `Type` because it contained the basic AST types?  Anyway, changing the name of both modules to `CoreTypes` or something might make sense.

Reply:
It's like `*.Function` is a module that stores functions over something. I.e. `*.Type` is something that stores types of something (i.e. ASTs, yes), not related to the `Type` type. I wouldn't mind changing the name to `Ast`, for example, but this is certainly for a different PR.